be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -459,52 +457,121 @@ Status JsonReader::_read_rows(Chunk* chunk, int32_t rows_to_read, int32_t* rows_
     return Status::OK();
 }
 
-Status JsonReader::_construct_row(simdjson::ondemand::object* row, Chunk* chunk,
-                                  const std::vector<SlotDescriptor*>& slot_descs) {
-    if (_scanner->_json_paths.empty()) {
-        // No json path.
-
-        for (SlotDescriptor* slot_desc : slot_descs) {
-            if (slot_desc == nullptr) {
+Status JsonReader::_construct_row_in_object_order(simdjson::ondemand::object* row, Chunk* chunk) {
+    // make a copy for _slot_desc_dict.
+    auto slot_desc_dict(_slot_desc_dict);
+    try {
+        std::ostringstream oss;
+
+        for (auto field : *row) {
+            // TODO:This is ugly, but there's no good way to convert json field key to std::string.
+            // convert field key to std::string.
+            simdjson::ondemand::raw_json_string raw_key = field.key();
+            oss << raw_key;
+            auto key = oss.str();
+            oss.str("");
+
+            // look up key in the slot dict.
+            auto itr = slot_desc_dict.find(key);
```

Comment:
Does the slot_desc_dict not use std::string, use char* can ok?
It can reduce the memory copy overhead

Reply:
If `char*` is the key type of `std::unoreded_map, the actual key is the pointer, not the pointee.